kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseSiteService.java

```diff
@@ -3702,30 +3702,32 @@ public void update(Observable o, Object eventObj) {
 
 		String eventType = event.getEvent();
 
-		if (EVENT_SITE_USER_INVALIDATE.equals(eventType) || PreferencesService.SECURE_EDIT_PREFS.equals(eventType)) {
-			// KNL-1171: always clear the cache for the user as the Site below may have been deleted
-			clearUserCacheForUser(event.getUserId());
+        if (eventType != null) {
+            switch (eventType) {
+                case SiteService.SECURE_UPDATE_SITE_MEMBERSHIP:
+                case SiteService.SECURE_UPDATE_GROUP_MEMBERSHIP:
+                case SiteService.SECURE_UPDATE_SITE:
+                    notifySiteParticipant("/gradebook/" + event.getContext() + "/");
+                    break;
+                case EVENT_SITE_USER_INVALIDATE:
+                    try {
+                        Site site = getSite(event.getResource());
+                        clearUserCacheForSite(site);
+                    } catch (IdUnusedException iue) {
+                        log.warn("Site not found when handling event ({})", event);
+                    }
```

Comment:
Do you need a `break` in this case? Or is the intention for it to flow into the `PreferencesService.SECURE_EDIT_PREFS` case as well?

Reply:
to let it flow!